docs/spec/connection-handshake/L2-tla/ICS3Module.tla

```diff
@@ -242,7 +252,7 @@ HandleTryMsg(m) ==
         myClientProof == GetClientProof
         replyMsg == [parameters |-> FlipConnectionParameters(m.parameters),
                      type |-> "ICS3MsgAck",
-                     proofHeight |-> store.latestHeight,
+                     proofHeight |-> store.latestHeight + 1,
```

Comment:
Can you explain why it's +1 here? I guess processing a message requires to increase chain height, but it is confusing seeing +1 here?Maybe we can increase chain height in `ProcessMsg` and then have actions for each handler just read the latest height?

Reply:
This `+1` is a fix for a [bug that Anca found](https://github.com/informalsystems/ibc-rs/pull/73#discussion_r440705958).

Explanation: We use `+1` to reflect that the chain advances its height before the proof is created. We advance the chain height via `NewStore`: 

https://github.com/informalsystems/ibc-rs/blob/eb0f21019778c0bf40f767d59f8d2f2002587ead/docs/spec/connection-handshake/L2-tla/ICS3Module.tla#L192-L194
that is part of every `Handle*` action, at the same time with computing `proofHeight |-> store.latestHeight + 1`. In the current spec we cannot do these two steps (advancing the height in `NewStore`, and computing the `proofHeight`) separately, so we do them at the same time, and that's why we see this `+1` in two places.

Reply:
> Maybe we can increase chain height in ProcessMsg and then have actions for each handler just read the latest height?

Not sure how to acheive this. It would require that we split the `Next` action of the ICS3 module into two parts, one for increasing the height & processing a message, and another for computing proof height. This could work but it's a more major modification.

Reply:
We shouldn't be increasing height as part of message handlers (including NewStore), but outside. If there is a message to be processed then we increase height and then process message. Proof height is always the latest height, i.e., the height at the moment you are reading the state. I don't get why you need to split next action into two parts.

Reply:
> If there is a message to be processed then we increase height and then process message. 

Note that all three actions happen in the same TLA+ step: (i) increasing the height, (ii) processing the message, and (iii) creating the proof. To do it otherwise than this, the `Next` action of the module would need to have separate high-level actions for each of these. (Major changes would be required for that.)

After a sync with @ancazamfir we made some small modifications that might make it better. WDYT? See [here](https://github.com/informalsystems/ibc-rs/pull/109/commits/1d988dc7dcdd14302003fc03341601daa357ff44).